app/containers/wallet/VotingPage.js

```diff
@@ -38,190 +20,36 @@ type Props = {|
-    ]);
-    const { intl } = this.context;
-    return (
-      <div>
-        {this.getDialog()}
-        <SpendingPasswordInput
-          classicTheme={false}
-          setForm={form => this.setSpendingPasswordForm(form)}
-          isSubmitting={false}
-        />
-        <Button
-          className={buttonClasses}
-          label={intl.formatMessage(globalMessages.nextButtonLabel)}
-          onMouseUp={this._vote}
-          skin={ButtonSkin}
-        />
-        <div>
-          {this.generated.stores.voting.encryptedKey !== null ? (
-            <QRCode
-              value={this.generated.stores.voting.encryptedKey}
-              bgColor={qrCodeBackgroundColor}
-              fgColor={qrCodeForegroundColor}
-              size={152}
-            />
-          ) : (
-            ''
-          )}
-        </div>
-      </div>
-    );
-  }
-
-  getDialog: void => void | Node = () => {
-    const { intl } = this.context;
-    const { votingRegTransaction } = this.generated.stores.substores.ada;
-    const votingRegTx = votingRegTransaction.createVotingRegTx.result;
-
-    const selectedWallet = this.generated.stores.wallets.selected;
-    if (selectedWallet == null) {
-      return null;
-    }
-
-    const showSignDialog =
-      this.generated.stores.wallets.sendMoneyRequest.isExecuting ||
-      !this.generated.stores.wallets.sendMoneyRequest.wasExecuted ||
-      this.generated.stores.wallets.sendMoneyRequest.error != null;
-
-    if (votingRegTransaction.createVotingRegTx.isExecuting) {
-      return (
-        <Dialog
-          title={intl.formatMessage(globalMessages.processingLabel)}
-          closeOnOverlayClick={false}
-        >
-          <AnnotatedLoader
-            title={intl.formatMessage(globalMessages.processingLabel)}
-            details={intl.formatMessage(globalMessages.txGeneration)}
-          />
-        </Dialog>
-      );
-    }
-
-    if (votingRegTransaction.createVotingRegTx.error != null) {
-      return this._errorDialog(votingRegTransaction.createVotingRegTx.error);
-    }
-    if (votingRegTx != null && showSignDialog) {
-      return (
-        <VotingRegTxDialog
-          staleTx={votingRegTransaction.isStale}
-          transactionFee={votingRegTx.fee()}
-          isSubmitting={this.generated.stores.wallets.sendMoneyRequest.isExecuting}
-          isHardware={
-            selectedWallet.getParent().getWalletType() === WalletTypeOption.HARDWARE_WALLET
-          }
-          getTokenInfo={genLookupOrFail(this.generated.stores.tokenInfoStore.tokenInfo)}
-          onCancel={this.cancel}
-          onSubmit={({ password }) =>
-            this.generated.actions.ada.votingTransaction.signTransaction.trigger({
-              password,
-              publicDeriver: selectedWallet,
-            })
-          }
-          classicTheme={this.generated.stores.profile.isClassicTheme}
-          error={this.generated.stores.wallets.sendMoneyRequest.error}
+    const { uiDialogs } = this.generated.stores;
```

Comment:
Remember you need to disable this flow for hardware wallets

Reply:
Also you need to disable this for Byron wallets (AKA wallets with no signing key)